docs/topics/testing/tools.txt

```diff
@@ -1194,11 +1194,6 @@ easy::
             with self.modify_settings(MIDDLEWARE={
                 'append': 'django.middleware.cache.FetchFromCacheMiddleware',
                 'prepend': 'django.middleware.cache.UpdateCacheMiddleware',
-                'remove': [
-                    'django.contrib.sessions.middleware.SessionMiddleware',
-                    'django.contrib.auth.middleware.AuthenticationMiddleware',
-                    'django.contrib.messages.middleware.MessageMiddleware',
-                ],
```

Comment:
@timgraham - I think you referred to if this should stay or go. IMO, if it stays, then we need to explain that it's okay because the same values aren't both removed and added. The note further down already does this, but I don't think it fully applies because it's so far down... better to refine the example, so it doesn't promote a behavior that will cause bugs that are hard to find and understand.

I think the example of a test with caching may still seem understandable to the user. I'm also not quite sure why the other middleware is removed... if something isn't fetched from cache, then the behavior of pages without authentications and sessions seem eh undefined. I wouldn't do that :)

Reply:
I think this is now an unrelated change and needs to be removed. 

(With everything in the `note::` we can leave the existing examples as they are.)

Reply:
I'm more +1 of removing the "bad example" and +0 on writing a long note. Reading the above and having never questioned dictionary key order, I find it likely that I would mix remove/append/prepend and not think about it.. *because the example did the same*.

Reply:
I think we're spending a lot of effort here for a problem that only affects Python 3.5 (based on what versions Django currently supports). Would you keep this note once support for Python 3.5 is dropped?

Reply:
Worth noting that even with Python 3.6 support, the order of remove/append/prepend matters, so the note could possible be kept + shortened if adapted to Python 3.6+ to communicate that the order matters?

Reply:
My preference is to make the note for Python 3.6+ most prominent and then have a self-contained block for Python 3.5 that could easily be removed.